kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -61,6 +74,145 @@ def __init__(self, *args, **kwargs):
+        compute: Optional[bool] = False,
+    ) -> Union[EBSD, LazyEBSD]:
+        """Creates a dictionary of EBSD patterns for a sample in the
+        (RD, TD, ND) reference frame, given a set of local crystal lattice
+        rotations and a detector model from a master pattern in the Lambert
+        projection.
+
+        Parameters
+        ----------
+        rotations : Rotation
+            Set of unit cell rotations to get patterns from.
+        detector : EBSDDetector
+            EBSDDetector object describing the detector geometry with one
+            projection center.
+        energy : int
+            The wanted energy in the master pattern.
+        n_chunk : int, optional
+            The number of chunks the data should be split up into. By default,
+            this is set so each chunk is around 100 MB.
+        dtype_out : numpy.dtype, optional
+            Data type of the returned patterns, by default np.float32.
+        compute : bool, optional
+            Wheter or not the dask.compute() function should be called, by
+            default false.
+            For more information visit the dask documentation at:
+            https://docs.dask.org/en/latest/api.html#dask.compute
+
+        Returns
+        ----------
+        this : EBSD or LazyEBSD
+            All the simulated EBSD patterns with the shape (number of rotations,
+            detector pixels in x direction, detector pixels in y direction).
+
+        Notes
+        ----------
+        If the master pattern phase has a non-centrosymmetric point group, both
+        the northern and southern hemispheres must be provided.
+        For more details regarding the reference frame visit the reference frame
+        user guide at: https://kikuchipy.org/en/latest/reference_frames.html.
+        """
+
+        if self.projection != "lambert":
+            raise NotImplementedError(
+                "Method only supports master patterns in lambert projection!"
+            )
+        pc = detector.pc_emsoft()
+        if len(pc) > 1:
+            raise ValueError("Method only supports a single projection center!")
+
+        # 4 cases
+        # Has energies, has hemis - Case 1
+        if len(self.axes_manager.shape) == 4:
+            energies = self.axes_manager["energy"].axis
+            energy_index = (np.abs(energies - energy)).argmin()
+            mpn = self.data[0, energy_index]
+            mps = self.data[1, energy_index]
+        # no energies, no hemis - Case 2
+        elif len(self.axes_manager.shape) == 2:
+            if not self.phase.point_group.contains_inversion:
+                raise HemisphereError(
+                    "For phases without inversion symmetry, "
+                    "both hemispheres must be in master pattern!"
+                )
+            mpn = self.data
+            mps = mpn
+        else:
+            try:  # has energies, no hemi - Case 3
+                if not self.phase.point_group.contains_inversion:
```

Comment:
Since this check is called twice, I think we should separate it out as it's own private method, however, I will do this after this PR is in, if that's OK with you @friedkitteh. (You can resolve this after you've read it.)